bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/Rule.java

```diff
@@ -76,19 +85,20 @@ public Rule(String uid) {
      *            and the template is not resolved.
      * @param visibility visibility of rule
      */
-    public Rule(String uid, List<Trigger> triggers, //
-            List<Condition> conditions, //
-            List<Action> actions, //
-            List<ConfigDescriptionParameter> configDescriptions, //
-            Configuration configurations, String templateUID, Visibility visibility) {
-        this.uid = uid;
-        setTriggers(triggers);
-        setConditions(conditions);
-        setActions(actions);
-        setConfigurationDescriptions(configDescriptions);
-        setConfiguration(configurations);
+    @SuppressWarnings("null")
+    public Rule(@Nullable String uid, @Nullable List<Trigger> triggers, @Nullable List<Condition> conditions,
+            @Nullable List<Action> actions, @Nullable List<ConfigDescriptionParameter> configDescriptions,
+            @Nullable Configuration configurations, @Nullable String templateUID, @Nullable Visibility visibility) {
+        this.uid = uid == null ? ID_PREFIX + UUID.randomUUID().toString() : uid;
+        this.triggers = triggers == null ? Collections.<Trigger> emptyList() : triggers;
+        this.conditions = conditions == null ? Collections.<Condition> emptyList() : conditions;
+        this.actions = actions == null ? Collections.<Action> emptyList() : actions;
+        this.configDescriptions = configDescriptions == null ? new ArrayList<ConfigDescriptionParameter>()
+                : configDescriptions;
+        this.configuration = configurations == null ? new Configuration() : configurations;
         setTemplateUID(templateUID);
-        setVisibility(visibility);
+        this.visibility = visibility == null ? Visibility.VISIBLE : visibility;
+        tags = Collections.<String> emptySet();
```

Comment:
Shouldn't `Collections.emptySet()` already do its job? The type could be identified implicit.

Reply:
> Shouldn't Collections.emptySet() already do its job? The type could be identified implicit.

Yes, you are correct. This was copy-pasted from old, java5 compatible code. I'll fix it

> I also didn't check if every null warning suppression is really necessary. Do we need all of them?

All null warning suppressions were added because of Eclipse warnings. I'll review them again